### DISCUSSION TITLE
error with the sktime autoarima function

### DISCUSSION THREAD

**Discussion by Abhaysharma53** [alan-turing-institute/sktime#3387] [Q&A]
Hello, I am really new to sktime and i am trying to create a reusable script using sktime's autoarima. but i am facing an error that i am not able to understand properly. it will be really helpful if anyone can help and let me know why i am facing this error and how cani remove that.
Error - KeyError: "None of [Int64Index([0, 1, 2, 3], dtype='int64')] are in the [index]"
A bit about the script - this script takes 2 parameter, dataframe and the forecasting horizon and returns the plot .
Thanks in advance.

**Comment by fkiraly** [comment]
This is a known bug coming from incompatibility of recently released `pmdarima 2.0.0` with earlier `sktime` versions, see here:
https://github.com/alan-turing-institute/sktime/issues/3300

It should work if you upgrade to `sktime 0.13.2` as it will ensure that `pmdarima 1.X` will be used.

A fix of the bug itself, compatible with the most recent `pmdarima` versions, will be in the next release.

**Comment by Abhaysharma53** [comment]
Hi @fkiraly ,
you are right, but this is not exactly true, so earlier i was already using sktime 0.13.2 and i had pmdarima in my system 2.0.1, and i was facing this issue. then i compared and got to know this, so i  had to manually downgrade my pmdarima version to 1.8.5, now it's working in my jupyter notebook but still i am not able to fix my issue.
so for my usecase, i am actually working on streamlit with sktime , now even though sktime code  is currently working for me in notebook , in my web application environment, it's giving me another issue now which is with respect to the parameter of sktime.
Error - __init__() got an unexpected keyword argument 'sp'
if i am trying to remove this parameter, i am getting same error with respect to suppress_warning and n_jobs, if i'm removing all these parameter i am getting another error wrt Failed in nopython mode pipeline
I am not able to understand why the same snippet is working for me in notebook whereas in application it's showing an error wrt model parameter. Any help onto this will be highly appreciated.

**Reply by fkiraly** [reply]
odd, no idea - can you reproduce the condition for your error? Do you have some code you can share?

High-level, sounds like accidentally parameters get passed directly to `pmdarima` objects rather than `sktime` - `sp` is one of the very few parameters we renamed between the two, to have consistenty between estimators.
